models/iterable__events.sql

```diff
@@ -0,0 +1,84 @@
+{{
+    config(
+        materialized='incremental',
+        unique_key='event_id',
+        partition_by={
+            "field": "created_on",
+            "data_type": "date"
+        } if target.type == 'bigquery' else none,
+        incremental_strategy = 'merge',
+        file_format = 'delta'
+    )
+}}
+
+with events as (
+
+    select *
+    from {{ var('event') }}
+
+    {% if is_incremental() %}
+    where created_at >= (select max(created_at) from {{ this }} )
+    {% endif %}
+
+), campaign as (
+
+    select *
+    from {{ ref('int_iterable__recurring_campaigns') }}
```

Comment:
i'm wondering if it makes more sense to join with campaign history and the state of the campaign at the time of event occurrence, rather than the current state? or maybe both?